### PR TITLE
[FLINK-16406] Increase default value for JVM Metaspace to minimise its OutOfMemoryError

### PR DESCRIPTION
**Pull request by azagrebin**
With FLIP-49 (#9801), we introduced a limit for JVM Metaspace ('taskmanager.memory.jvm-metaspace.size') when TM JVM process is started. It caused `OutOfMemoryError: Metaspace` for some use cases after upgrading to the latest 1.10 version. In some cases, a real class loading leak has been discovered. Some users had to increase the default value to accommodate for their use cases (mostly from `96Mb` to `256Mb`).

While this limit was introduced to properly plan Flink resources, especially for container environment, and to detect class loading leaks, the user experience should be as smooth as possible. In general, the size depends on the use case (job user code, how many jobs are deployed in the cluster etc).

This issue tries to tackle this problem by firstly increasing it to `256Mb` and overall default process size to `1728Mb` in `flink-conf.yaml` to have no impact on default sizes of other memory components.

### CODE REVIEW

File: flink-dist/src/main/resources/flink-conf.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobmanager.heap.size: 1024m
 #
 # Note this accounts for all memory usage within the TaskManager process, including JVM metaspace and other overhead.
 
-taskmanager.memory.process.size: 1568m
+taskmanager.memory.process.size: 1728m
```
Comment (zentol) [inline flink-dist/src/main/resources/flink-conf.yaml]: is this fully derived from existing config options? Maybe we could generate parts of the `flink-conf.yaml`

Comment (azagrebin) [reply flink-dist/src/main/resources/flink-conf.yaml]: I would not say so. Every time it is a custom decision.

**Comment by xintongsong** [comment]
Thanks for the PR @azagrebin. LGTM.
I'm wondering shall we hold this PR for now? Just to give some time for people responding the the user ML survey. I guess there is still time before releasing 1.10.1?

**Comment by azagrebin** [comment]
Thanks for the review @tillrohrmann @xintongsong 
True, I also think we should wait for couple of days to give users a chance for some feedback before including this to 1.10.1.

**Comment by azagrebin** [comment]
@flinkbot run travis

**Comment by azagrebin** [comment]
@flinkbot run azure

**Comment by rmetzger** [comment]
I think the e2e test failure on azure is a real regression proposed in this PR, as it affects the memory management e2e test
```
[] - The derived from fraction jvm overhead memory (102.400mb (107374184 bytes)) is less than its min value 192.000mb (201326592 bytes), min value will be used instead
[] - The derived from fraction network memory (57.600mb (60397978 bytes)) is less than its min value 64.000mb (67108864 bytes), min value will be used instead
Exception in thread "main" org.apache.flink.configuration.IllegalConfigurationException: Sum of configured Framework Heap Memory (128.000mb (134217728 bytes)), Framework Off-Heap Memory (128.000mb (134217728 bytes)), Task Off-Heap Memory (0 bytes), Managed Memory (300.000mb (314572800 bytes)) and Network Memory (64.000mb (67108864 bytes)) exceed configured Total Flink Memory (576.000mb (603979776 bytes)).
	at org.apache.flink.runtime.clusterframework.TaskExecutorProcessUtils.deriveInternalMemoryFromTotalFlinkMemory(TaskExecutorProcessUtils.java:320)
	at org.apache.flink.runtime.clusterframework.TaskExecutorProcessUtils.deriveProcessSpecWithTotalProcessMemory(TaskExecutorProcessUtils.java:248)
	at org.apache.flink.runtime.clusterframework.TaskExecutorProcessUtils.processSpecFromConfig(TaskExecutorProcessUtils.java:147)
	at org.apache.flink.runtime.util.BashJavaUtils.getTmResourceJvmParams(BashJavaUtils.java:62)
	at org.apache.flink.runtime.util.BashJavaUtils.main(BashJavaUtils.java:46)
[ERROR] Could not get JVM parameters properly.
```

**Comment by azagrebin** [comment]
@flinkbot run travis

**Comment by azagrebin** [comment]
@flinkbot run azure

**Comment by azagrebin** [comment]
@flinkbot run travis

**Comment by azagrebin** [comment]
@flinkbot run azure

**Comment by azagrebin** [comment]
Thanks @rmetzger. Indeed, it was memory misalignment in that e2e test. I have fixed it.

**Comment by DestyNova** [comment]
Thanks for this. I've been encountering OOMs on metaspace with 1.10 too, and was quite confused by the documentation which [states a 256MB default value](https://ci.apache.org/projects/flink/flink-docs-stable/ops/config.html) when actually it seems to be 96MB.

**Comment by xintongsong** [comment]
@DestyNova 

I assume you are using Flink 1.10.0? This PR has been merged after the release of 1.10.0. It should be included in release 1.10.1 & release 1.11.0, both of which are not released yet.

Regarding the documentation, I think it's built based on the 1.10 snapshot version, which is a bit ahead of the stable releases. This is indeed a problem, I'll try to find people to fix it. Thanks for reporting this.

**Comment by azagrebin** [comment]
@DestyNova @xintongsong 
For the record:
https://issues.apache.org/jira/browse/FLINK-17551

**Comment by DestyNova** [comment]
@xintongsong @azagrebin thanks for considering the problem.

Incidentally, I'm still getting metaspace OOMs with 512mb max metaspace size. I'm guessing something is generating lots of anonymous proxy classes at runtime, but I'm not sure. Are there good ways to debug this? The Flink dashboard UI gives some information about task manager heap etc, but doesn't seem to say anything about the metaspace situation or e.g. how many classes are loaded.

**Comment by xintongsong** [comment]
@DestyNova,
Flink does not provide class loading/unloading tracing. However, you can do that by manually specifiying JVM options. You can add `-XX:+TraceClassLoading -XX:+TraceClassUnloading` to the Flink configuration option `env.java.opts.taskmanager`. Then you should see information about loading/unloading classes in your GC log.

**Pull request closed without merging**